From 81c175153aa8c36af3368c04d1cdf089c05a941f Mon Sep 17 00:00:00 2001
From: Pavel Valena <pvalena@redhat.com>
Date: Fri, 6 Mar 2026 13:09:32 +0100
Subject: [PATCH] boot/dracut: use systemdsystemunitdir instead of
 systemdsystemconfdir

since dracut-109 [*], the systemdsystemconfdir exists only in hostonly mode, which leads to unbootable system; as documented here:
https://src.fedoraproject.org/rpms/dracut/pull-request/90

[*] https://github.com/dracut-ng/dracut-ng/commit/2991f74a
---
 src/boot/dracut/module-setup.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/boot/dracut/module-setup.sh b/src/boot/dracut/module-setup.sh
index 06d74600..4a70a2ee 100755
--- a/src/boot/dracut/module-setup.sh
+++ b/src/boot/dracut/module-setup.sh
@@ -46,7 +46,7 @@ install() {
         inst_simple "/etc/ostree/initramfs-root-binding.key"
     fi
     inst_simple "${systemdsystemunitdir}/ostree-prepare-root.service"
-    mkdir -p "${initdir}${systemdsystemconfdir}/initrd-root-fs.target.wants"
+    mkdir -p "${initdir}${systemdsystemunitdir}/initrd-root-fs.target.wants"
     ln_r "${systemdsystemunitdir}/ostree-prepare-root.service" \
-        "${systemdsystemconfdir}/initrd-root-fs.target.wants/ostree-prepare-root.service"
+        "${systemdsystemunitdir}/initrd-root-fs.target.wants/ostree-prepare-root.service"
 }
-- 
2.47.3

